Post-upmerge followup work (#5349)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

    * The library has been deprecated in favor of the :ref:`at_monitor_readme` library.

  * Board names:

>     * The ``nrf9160dk_nrf9160ns`` and the ``nrf5340dk_nrf5340_cpuappns`` boards have been renamed respectively to ``nrf9160dk_nrf9160_ns`` and ``nrf5340dk_nrf5340_cpuapp_ns``, in a change inherited from upstream Zephyr.

Reviewer: ```suggestion
    * The ``nrf9160dk_nrf9160ns`` and the ``nrf5340dk_nrf5340_cpuappns`` boards have been renamed respectively to ``nrf9160dk_nrf9160_ns`` and ``nrf5340dk_nrf5340_cpuapp_ns``, in a change inherited from the upstream Zephyr.
```

MCUboot
=======

The MCUboot fork in |NCS| (``sdk-mcuboot``) contains all commits from the upstream MCUboot repository up to and including ``7a51968``, plus some |NCS| specific additions.

The code for integrating MCUboot into |NCS| is located in :file:`ncs/nrf/modules/mcuboot`.

The following list summarizes the most important changes inherited from upstream MCUboot:

> * added AES support for image encryption (mbed TLS based).

Reviewer: ```suggestion
* Added AES support for image encryption (based on mbedTLS).
```

> * boot serial: Ported encoding to use cddl-gen (which removes dependency on TinyCBOR).

Reviewer: ```suggestion
* MCUboot serial: Ported encoding to use the cddl-gen module (which removes dependency on the `TinyCBOR`_ library).
```
Isn't this MCUboot serial? Also, is this related to the CONFIG_CDDL_GEN Kconfig?

> * bootutil_public: Made boot_read_swap_state() declaration public.

Reviewer: ```suggestion
* bootutil_public library: Made :c:func:`boot_read_swap_state()` declaration public.
```

Zephyr
======

.. NOTE TO MAINTAINERS: All the Zephyr commits in the below git commands must be handled specially after each upmerge and each NCS release.

The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``14f09a3b00``, plus some |NCS| specific additions.

For a complete list of upstream Zephyr commits incorporated into |NCS| since the most recent release, run the following command from the :file:`ncs/zephyr` repository (after running ``west update``):

.. code-block:: none

   git log --oneline 14f09a3b00 ^v2.6.0-rc1-ncs1

For a complete list of |NCS| specific commits, run:

.. code-block:: none

   git log --oneline manifest-rev ^14f09a3b00

The current |NCS| master branch is based on the Zephyr v2.7 development branch.

The following list summarizes the most important changes inherited from upstream Zephyr:

> .. TODO update the following sections to reflect https://github.com/nrfconnect/sdk-nrf/pull/5189. Delete this comment once that is complete.

Reviewer: @mbolivar-nordic , correct me if I'm wrong, but I remember a conversation with @carlescufi and @shantha-14 around the NCS 1.6.0 release where it was decided that we would only link to the Zephyr release notes and skip listing all the changes. So these sections should be removed and we should just have a link to https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/releases/release-notes-2.7.html , with potentially a subsection detailing more complex changes, like in https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/releases/release-notes-1.6.0.html#zephyr : 
```suggestion
See the :ref:`zephyr:zephyr_2.7` Release Notes for an overview of the most important changes inherited from upstream Zephyr up to the commit ``14f09a3b00``.
```
